src/hooks/connectors/useMetamask.ts

```diff
@@ -40,5 +40,22 @@ export function useMetamask() {
     "Metamask connector not found, please make sure it is provided to your <ThirdwebProvider />",
   );
 
-  return () => connect(connector);
+  return () => {
+    if (typeof window !== "undefined") {
+      // browser context
+      if (!window.ethereum) {
+        // no injected connector available
+        window.open(
+          `https://metamask.app.link/dapp/${
```

Comment:
let's do the walletconnect thing here instead

Reply:
yup!